engine/flink/api/src/main/scala/pl/touk/nussknacker/engine/flink/util/context/FlinkContextInitializer.scala

```diff
@@ -0,0 +1,36 @@
+package pl.touk.nussknacker.engine.flink.util.context
+
+import org.apache.flink.api.common.functions.MapFunction
+import pl.touk.nussknacker.engine.api.Context
+import pl.touk.nussknacker.engine.api.context.ProcessCompilationError.NodeId
+import pl.touk.nussknacker.engine.api.context.ValidationContext
+import pl.touk.nussknacker.engine.api.typed.typing
+
+/**
+  * FlinkContextInitializer provides definition of Context returned within DataStream by FlinkSource
+  * and map function that transforms raw input to Context.
+  *
+  * @tparam T - type of event that is generated by raw source.
+  */
+abstract class FlinkContextInitializer[T] extends Serializable {
+
+  def validationContext(context: ValidationContext, name: String, result: typing.TypingResult)(implicit nodeId: NodeId): ValidationContext
```

Comment:
1. Why we need this `name` here? Is it a name in case when we produce only one variable?
2. Why we need `result` here? What is it?
3. Shouldn't both be passed as a parameters of constructor in very specific case whene source produces only one variable?

Reply:
1 & 2. Changed name -> outputVariableName and result -> outputVariableType and fixed documentation. Those parameters represent default one variable produced by the source.
3. For multiple variables see `GenericSourceWithCustomVariablesSample`. There is one `val customContextInitilalizer` which is used in two places:
- creating the source
- contextTransformation - This is the place where I know all variables and their types (especially in avro sources), and pass them as parameters to the method of already created object. 
I see two alternatives:
1. pass a List of variable name-and-type pairs
2. initialize `var customContextInitilalizer` with new FlinkContextInitializer inside contextTransformation
WDYT?

Reply:
Ok, I see now. WDYT about passing to this `validationContext` method everything what we have in `TransformationStep`? I know that there are type dependent types but we can have `parameters: List[(String, BaseDefinedParameter)], state: Option[Any]` and someone could do some pattern matching on it. WDYT? Thanks to that we will have the whole power of `GenericNodeTransformation` in this place.